.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,39 @@
+name: e2e-mock-provider-tests
+on: 
+  workflow_dispatch:
+    inputs:
+      registry:
+        description: 'Registry'
+        required: true
+        default: 'gcr.io/k8s-staging-csi-secrets-store'
+      driverImage:
+        description: 'Secret Store CSI driver image'
+        required: true
+        default: 'driver'
+      driverImageVersion:
+        description: 'Secret Store CSI driver image version'
+        required: true
+        default: 'v0.3.0'
+jobs:
+  e2e-test:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+          fetch-depth: 0  
```

Comment:
Should we also setup go and pin to a version here? We're using `1.17` and I assume if not explicitly pinned it'll use the default on the agent?

```yaml
      - uses: actions/setup-go@v2
        with:
          go-version: "^1.17"
```